arduino/SLAVE_I2C/SLAVE_I2C.ino

```diff
@@ -61,6 +71,280 @@ void loop() {
       setColor(0, 255, 0);
       break;
   }
+    
```

Comment:
you can simplify these loops greatly... you're never going to get something that's not a integer, so you can use the position as part of your loop, as I mentioned earlier. don't worry about saving it to a variable, you can set state from `receiveEvent`

in addition, can you get rid of previous switch case for the time being? your elevator values are going to overlap with those.

Reply:
For the loops, why is it always an integer that's received? Most of the time, it won't be, unless we're talking about different values?

Also, isn't the switch case code for another LED strip, so why would our elevator meter values interfere with those?

Reply:
because of how I2C works, you're converting to integer math before being sent... check Arduino.java.. I think there's some casts you're overlooking there. Java numbers are slightly different than JavaScript, something to remember. 

although the switch case code is for another LED strip, you're using overlapping values... if you send the value 6, you're going to trigger the enable state and 6/30 on your elevator meter. in addition, that particular case has delays in it, so it'll delay the meter output for 0.5s approx.

Reply:
See line 176: https://github.com/FRCteam4909/BionicFramework/pull/135/commits/c92ae39b68f851caca36dd103c0baf7fad9cdba6#diff-0b01612aebfe60036ff49aa834c80afaR176

actually, you're ~casting~ doing your calculation as ints,  which is going to result in really funky number, again not JS... see [some StackOverflow goodness]( https://stackoverflow.com/questions/9102318/cast-double-to-integer-in-java)

Reply:
yeah, your fix should tackle most of the issues w/ the sending the number, just that you're truncating w/ your cast not rounding - that's minimal.

however, you still want to get rid of the switch case for the other strip because when it receives a 6, it's going to try to trigger both events, one of which also causes a delay. These numbers shouldn't overlap.

Reply:
Okay, should I get rid of it completely? Because then how would we send the code for the other strip, just use another arduino?

Reply:
one way to solve that would be to use numbers that don't overlap, and to not do delays. say we send 0-30 for our elevator, send something 31+ for the other strip